src/picongpu/include/fields/currentDeposition/ZigZag/ZigZag.hpp

```diff
@@ -316,12 +325,10 @@ struct ZigZag
     DINLINE float_X
     calc_relayPoint(const float_X i_1, const float_X i_2, const float_X x_1, const float_X x_2) const
     {
-
-        const float_X min_1 = ::min(i_1, i_2) + float_X(1.0);
-        const float_X max_1 = ::max(i_1, i_2);
-        const float_X max_2 = ::max(max_1, (x_1 + x_2) / float_X(2.));
-        const float_X x_relayPoint = ::min(min_1, max_2);
-        return x_relayPoint;
+        /* paper version:
+         *   i_1 == i_2 ? (x_1 + x_2) / float_X(2.0) ::max(i_1, i_2);
+         */
+        return i_1 == i_2 ? x_2 : ::max(i_1, i_2);
```

Comment:
Currently libPMacc has no `max` implementation for `float` and `double` therefor the default cuda max/min is used.
The libPMacc min/max is coming in a separate pull request.